Sources/FuntastyKit/Architecture/Coordinator.swift

```diff
@@ -32,8 +32,10 @@ public protocol DefaultCoordinator: Coordinator {
 }
 
 public protocol ShowCoordinator: DefaultCoordinator {
-    var sourceViewController: UIViewController { get }
+    /// sourceViewController should be used as 'weak var', otherwise memory leak will emerge
+    var sourceViewController: UIViewController? { get }
```

Comment:
```suggestion
    var sourceViewController: UIViewController! { get }
```

Reply:
As discussed, in this case all protocols should be updated as all projects that are using FuntastyKit coordinators. So for the sake of consistency and compatibility, it must be optional.